src/lib/api.ts

```diff
@@ -11,6 +11,11 @@ import http from 'http'
 import https from 'https'
 import { hasReadPermissions } from './utils'
 import { getKubeAPIServiceAccountToken, getNamespace, initHTTPSTrustStore } from './kube'
+import { execSync } from 'node:child_process'
+
+export const getGitCommitHash = () => {
+    return execSync('git rev-parse HEAD').toString().trim()
```

Comment:
Do we need to make a change to `Dockerfile` so `.git` is there for this command to work from inside of an image? Should we have a fallback of some kind in case this function is not able to resolve a sha?

Reply:
yeah this was my only unknown, im flying a little blind in terms of what will be available in the deployed/containerized instance 🤔

Reply:
You can pull the latest production image `337909745635.dkr.ecr.us-east-1.amazonaws.com/setup-app:20250502-6ed7132b` locally if you want to inspect it, but from a cursory look I see at least two problems right now:

1. The image doesn't include `git` as a command
2. The image doesn't include `.git`

Now, I think the reason for item 2 is because the `.git` may not be there in the context when CodePipeline clones the repo. If that hypothesis is correct, there is probably some way to adjust that, but I have never run into this before so am not immediately aware of how 😅